### PR TITLE
build: fixed coverage regression introduced in #8957

### PR DESCRIPTION
**Pull request by d4l3k**
@tamird @jordanlewis 
#8957 `set -euo pipefail` caused the upload-coverage script to fail.

https://teamcity.cockroachdb.com/viewLog.html?buildId=16669&buildTypeId=Cockroach_Coverage_Coverage&tab=buildLog

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8989)

### CODE REVIEW

**Comment by jordanlewis** [comment]
LGTM

File: build/upload-coverage.sh [hunk 1]
```diff
@@ -25,6 +25,8 @@ coverage_dir="${outdir}/coverage"
 coverage_profile="${coverage_dir}/coverage.out"
 coverage_mode=count
 
+line_count=""
```
Comment (tamird) [inline build/upload-coverage.sh]: this is only used inside `iterative_coverpkg` - can it be made a local?

**Comment by tamird** [comment]
This needs to go to master plz

**Comment by d4l3k** [comment]
@tamird We also need to cherry-pick https://github.com/cockroachdb/cockroach/commit/1f50b9675ee6e0a70b036c72d1425c746882a7c5 into master.

File: build/upload-coverage.sh (hunk 1 above)
Comment (d4l3k) [inline build/upload-coverage.sh]: Done.

**Comment by tamird** [comment]
feel free to add that commit to this PR

**Comment by d4l3k** [comment]
Added & rebased on top of master.

**Comment by tamird** [comment]
LGTM

**Pull request merged**